org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -1330,6 +1336,22 @@ public static Preferences createFrom(Map<String, Object> configuration) {
 		prefs.setChainCompletionEnabled(chainCompletionEnabled);
 		List<String> diagnosticFilter = getList(configuration, JAVA_DIAGNOSTIC_FILER, Collections.emptyList());
 		prefs.setDiagnosticFilter(diagnosticFilter);
+		Object associations = getValue(configuration, JAVA_CONFIGURATION_ASSOCIATIONS);
+		List<String> filesAssociations = new ArrayList<>();
+		if (associations instanceof Map map) {
+			try {
+				Map<String, String> element = map;
+				element.forEach((k, v) -> {
+					// Java LS only recognizes *.<extension>
+					if ("java".equals(v) && k.startsWith("*.") && k.length() > 2) {
```

Comment:
According to the Eclipse preference page for this, it can support exact file names ( not just *.doc, but report.doc). Is there any issue allowing those ?

Reply:
I have tried it, but it doesn't work in Eclipse. Only *.doc works.

Reply:
That's fine then. Maybe the docs are just wrong there.

Reply:
I have updated it - https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/3230/commits/d78868370cfeaf62d1a9571ddc019fe82cf7c875#diff-6cfe34444f5bad0b61941c3d0be419cd8b7ee062522c9894aa2de57f4e0e218dR1345